File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.java
class Solution {
    public int maxProfit(int[] prices) {
        int lp = 0, rp = 1;
        int profit = 0;

        while (rp < prices.length) {
            if(prices[lp] < prices[rp]) {
                int profitPt = prices[rp] - prices[lp];
                profit = Math.max(profitPt, profit);
            } else {
                lp = rp;
            }
            rp++;
        }
    return profit;       
    }
}

File: Data Structures & Algorithms/trapping-rain-water/submission-1.py
class Solution:
    def trap(self, height: List[int]) -> int:
        
        n = len(height)

        l = 0
        r = n-1

        lc = height[l]
        rc = height[r]

        water = 0
        while l<r:
            if lc < rc:
                l+=1
                lc = max(lc, height[l])
                water += lc - height[l]
            else:
                r -=1
                rc = max(rc, height[r])
                water += rc - height[r]

            print(l, r, water, lc, rc)
        
        return water


        